cli/src/update/unix.rs

```diff
@@ -83,116 +117,36 @@ impl ApplicationUpdater {
         }
     }
 
-    /// Locate the currently installed asset on the given host.
-    fn installed_asset(
-        &self,
-        prefix: Option<&str>,
-        asset_name: &str,
-    ) -> Result<PathBuf, std::io::Error> {
-        let mut current_bin = std::env::current_exe()?;
-        current_bin.pop();
-        if let Some(p) = prefix {
-            current_bin.push(p);
-        }
-        current_bin.push(asset_name);
-        Ok(current_bin)
-    }
-
-    /// Generic function for fetching data from Github.
-    async fn get_github<T, C, F>(&self, url: &str, f: C) -> Option<T>
-    where
-        F: Future<Output = Option<T>>,
-        C: Fn(Response) -> F,
-    {
-        // let q = move |r: i32| async move { r };
-
-        let client = Client::builder().user_agent("phylum-cli").build();
-
-        match client {
-            Ok(client) => {
-                let response = client.get(url).send().await;
-
-                match response {
-                    Ok(response) => f(response).await,
-                    Err(_) => None,
-                }
+    /// Check for an update by querying the Github releases page.
+    async fn get_latest_version(&self, prerelease: bool) -> Option<GithubRelease> {
+        async fn inner(github_uri: &str, prerelease: bool) -> anyhow::Result<GithubRelease> {
```

Comment:
I'd assume an inner function was chosen here solely for `async`? Async blocks should work better without having to put functions inside of functions:

```rust
        let release = async {
            if prerelease {
                let url = format!("{}/repos/phylum-dev/cli/releases", self.github_uri);
                let r = http_get(&url).await?;
                let releases = r.json::<Vec<GithubRelease>>().await?;
                // Use the first one in the list, which should be the most recent
                releases
                    .first()
                    .cloned()
                    .ok_or_else(|| anyhow::anyhow!("no releases found"))
            } else {
                let url = format!("{}/repos/phylum-dev/cli/releases/latest", self.github_uri);
                let r = http_get(&url).await?;
                Ok(r.json::<GithubRelease>().await?)
            }
        };

        match release.await {
            Ok(ver) => {
                log::debug!("Found latest version: {:?}", ver);
                Some(ver)
            },
            Err(e) => {
                log::warn!("Failed getting latest version: {:?}", e);
                None
            }
        }
```

The match block changes here are mostly a style suggestion, I think it improves consistency between the two different match arms.